### ISSUE TITLE
Allow to pass CustomEndpointAddress to ServiceBusClientOptions

### ISSUE THREAD

**Issue by scrocquesel-ml150** [Azure/azure-functions-dotnet-worker#2853]
### Description

We are using a docker container to emulate Azure Service Bus, while it works like a charm with Azure SDK Client, it seems Azure Function do not use the same client factory as `Microsoft.Extensions.Azure`.

In other words, we want the client to be create with

```
    var sbClient = new ServiceBusClient(configSection["fullyQualifiedNamespace"],
            new ServiceBusClientOptions()
            {
                TransportType = configSection["transportType"],
                CustomEndpointAddress = configSection["customEndpointAddress"],
            });
```

The factory in `Microsoft.Extensions.Azure` make use of reflection to bind the config section to the ServiceBusClientOptions.
Something similar is done in the [webjobs extension](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs) but it misses the `CustomEndpointAddress` property.

**Comment by jviau** [comment]
In dotnet isolated the service bus client used is in an entirely different process, so configuration via code won't be possible here. @JoshLove-msft is it possible to read this value from `IConfiguration` in the WebJobs extension? Or is it already supported?

**Comment by JoshLove-msft** [comment]
Yes, this is already supported in the WebJobs extension.

**Comment by jviau** [comment]
> Yes, this is already supported in the WebJobs extension.

@JoshLove-msft is it supported via configuration? Or is it code only?

**Comment by JoshLove-msft** [comment]
My mistake - the CustomEndpointAddress is not available in the webjobs extension via config or code. It would need to be added into the [ServiceBusOptions type](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusOptions.cs), and then consumed in the [builder extension](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs). Is the ask here to support it in the WebJobs extension?

**Comment by scrocquesel-ml150** [comment]
> My mistake - the CustomEndpointAddress is not available in the webjobs extension via config or code. It would need to be added into the [ServiceBusOptions type](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusOptions.cs), and then consumed in the [builder extension](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs). Is the ask here to support it in the WebJobs extension?

Yes, my request is to be able to pass the CustomEndpointAddress with the connection configuration section to the dotnet isolated service bus trigger.

**Comment by jviau** [comment]
Yes, the ask would be to support this in the WebJobs extension. This will cover both in-proc and isolated.

@scrocquesel-ml150 when supported, you will need to set it via `host.json` or an app setting / env variable. This is because in dotnet isolated, the host process loads the _WebJobs_ versions of the extensions, and that is what performs all interactions with the actual service bus.

**Comment by JoshLove-msft** [comment]
Sounds good, I will file this in the azure-sdk-for-net repo where the code lives.

**Comment by CloudCees** [comment]
A scenario we want to implement via the `CustomEndpointAddress` is connecting with a Function to a Service Bus behind an Azure Application Gateway. The WebProxy setting is not working and is causing a "Unable to connect to the remote server" exception.

**Comment by jviau** [comment]
@CloudCees does using `fullyQualifiedNamespace` via IConfguration work for you? Or setting the address in the connection string.

https://learn.microsoft.com/en-us/azure/azure-functions/functions-identity-based-connections-tutorial-2#connect-to-the-service-bus-in-your-function-app

**Comment by jviau** [comment]
Seeing the age of this issue and that the Azure SDK team elected to not implement the ask, this feature is out of scope for this repo (as I mentioned above)
